### ISSUE TITLE
Question about detection function

### ISSUE THREAD

**Issue by SeanChenxy** [amdegroot/ssd.pytorch#124]
Hi everyone:
  I'm confused about the following codes in `class Detect(Function):`
```
    flt = output.contiguous().view(num, -1, 5)
    _, idx = flt[:, :, 0].sort(1, descending=True)
    _, rank = idx.sort(1)
    flt[(rank < self.top_k).unsqueeze(-1).expand_as(flt)].fill_(0)
```
Does those code do some operations on `self.output`? Could you give me some suggestions?

**Comment by ShoufaChen** [comment]
I check this by following way:

1.   assign   `begin = out `
before this line : `flt = output.contiguous().view(num, -1, 5)`
2.  assign  `hhh = (begin == output).sum()`
before the last line: `rerutn out`

3. then I got the **hhh** value is 21000, which is equals to the number of the elements of the output.

So I guess the output value didn't change when do following:
```
    flt = output.contiguous().view(num, -1, 5)
    _, idx = flt[:, :, 0].sort(1, descending=True)
    _, rank = idx.sort(1)
    flt[(rank < self.top_k).unsqueeze(-1).expand_as(flt)].fill_(0)
```

Is it right? @amdegroot 
I don't understand why do this ? Thank you !

**Comment by dingjiansw101** [comment]
@ShoufaChen I also have question about the code. Agree, I think it didn't change the output value.

**Comment by FostorHUNT** [comment]
It changes the output value by filling value 0 at the position not ranking the top k.

You can treat flt just as a Pointer which points to output variable.

**Comment by ShoufaChen** [comment]
@FostorHUNT  I am sorry that I can't agree with you. In fact, use the `mask` (`torch.uint8`) as slice, the tensor will not share the memory.

**Comment by pengfeidip** [comment]
@ShoufaChen  You are right in version 0.4.1.... but we don't know the fact in the verion of author's code..
